### PR TITLE
Migrate driver tests to gtest

### PR DESCRIPTION
**Pull request by apatil19**
Migrate following tests from framework/test/driver to framework/test/gtest
1. LDAPFilterTest
2. ServiceRegistryTest

Signed-off-by: The MathWorks, Inc. aadityap@mathworks.com

### CODE REVIEW

File: framework/test/gtest/ServiceRegistryTest.cpp [hunk 1]
```diff
@@ -75,39 +93,32 @@ void TestMultipleServiceRegistrations(BundleContext context)
 
   std::vector<ServiceReference<ITestServiceA>> refs =
     context.GetServiceReferences<ITestServiceA>();
-  US_TEST_CONDITION_REQUIRED(
-    refs.size() == 2, "Testing for two registered ITestServiceA services")
+  ASSERT_EQ(refs.size(), 2);
```
Comment (Jeeyuen-Rick) [inline framework/test/gtest/ServiceRegistryTest.cpp:96]: Probably add diagnostic message as the original test.

File: framework/test/gtest/LDAPFilterTest.cpp [hunk 2]
```diff
@@ -88,3 +89,87 @@ TEST(LDAPFilter, DefaultConstructedMatch)
...
+  props.clear();
+  props["cn"] = std::string("Babs Jensen");
+  ASSERT_FALSE(ldap.MatchCase(props));
+
+  // parentheses
+  ldap = LDAPFilter("(prop=foo(bar))");
+  props.clear();
+  props["prop"] = std::string("foo(bar)");
+  ASSERT_TRUE(ldap.Match(props));
+}
```
Comment (pelliott-mathworks) [inline framework/test/gtest/LDAPFilterTest.cpp:175]: Just wondering if it's okay that there is no newline at the end of the file

**Review by kevinleeMW — APPROVED** [review]
Thanks for this, Aaditya!

**Review by jeffdiclemente — APPROVED** [review]

**Review by Jeeyuen-Rick — APPROVED** [review]
Thanks!

**Pull request merged**